Frontend/src/Appointments/Appointments.jsx

```diff
@@ -1,11 +1,73 @@
+import React, { useState, useEffect } from 'react';
+import Calendar from "./Calendar";
 import Scheduler from "./Scheduler";
 
 function Appointments() {
+    const [userFirstName, setUserFirstName] = useState('');
+    const [userProfilePicture, setUserProfilePicture] = useState('');
+
+    useEffect(() => {
+        const userId = JSON.parse(localStorage.getItem("userId"));
+        async function fetchUserData() {
+            try {
+                const response = await fetch(`http://localhost:3000/${userId}/dashboard/name/picture`, {
+                    method: 'GET'
+                });
+                if (!response.ok) {
+                    throw new Error('Failed to fetch user data');
+                }
+                const data = await response.json();
+                setUserFirstName(data.first_name);
+                const buffer = data.profile_picture.data;
+                const base64String = btoa(String.fromCharCode(...new Uint8Array(buffer)));
+                setUserProfilePicture(`data:image/jpeg;base64,${base64String}`);
+            } catch (error) {
+                console.error('Error fetching User Data:', error);
+            }
+        }
+        fetchUserData();
+    }, []);
```

Comment:
Are you purposely only wanting to run this effect only on mount? I noticed your dependency array is empty.